### DISCUSSION TITLE
error connecting to server due to GSSAPI negotiation

### DISCUSSION THREAD

**Discussion by suenalaba** [supabase/supabase#37818] [Questions]
# Bug report

- [ X] I confirm this is a bug with Supabase, not with my own application. 
- [ X ] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

When connecting via pooling the following error is thrown:
`psql: error: connection to server at "aws-0-ap-southeast-1.pooler.supabase.com" (52.77.146.31), port 6543 failed: received invalid response to GSSAPI negotiation: `

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

Previously when I first started the project a few days ago, this connection worked:
`psql -h aws-0-ap-southeast-1.pooler.supabase.com -p 6543 -d postgres -U postgres.rkijdoxezclpigxoeacp`
But suddenly, it doesn't.

Only this works:
`psql -h db.rkijdoxezclpigxoeacp.supabase.co -p 5432 -d postgres -U postgres`

Tried the connection URI method on my application with similar outcomes.

Tested that: `curl -6 https://ifconfig.co/ip` and `curl -4 https://ifconfig.co/ip` both return ip addresses.

## Expected behavior

Previously it is expected that connecting via `psql -h aws-0-ap-southeast-1.pooler.supabase.com -p 6543 -d postgres -U postgres.rkijdoxezclpigxoeacp` worked. 

## Screenshots

Most of the settings are at default.
![Screenshot 2024-10-12 at 11 57 40 AM](https://github.com/user-attachments/assets/6d18da5a-b32f-4175-bdf3-d0863cefc963)
![Screenshot 2024-10-12 at 11 57 45 AM](https://github.com/user-attachments/assets/3bd80f7d-42c3-40f7-9035-296bb7b3d931)
![Screenshot 2024-10-12 at 11 57 50 AM](https://github.com/user-attachments/assets/d1ad8a5d-e951-4e97-8d75-1e189fce13d6)
![Screenshot 2024-10-12 at 11 57 58 AM](https://github.com/user-attachments/assets/4ad17012-59d3-4bf1-9212-9a3896e82eef)

## System information

- OS: macOS
- Browser: Chrome / zshell
- psql (PostgreSQL) 17.0 (Homebrew)

**Comment by acylicsalmon** [comment]
i have the same issue

**Comment by Matth5050** [comment]
same here

**Comment by pranavbadami** [comment]
If anyone ran into this, I found this [SO post](https://stackoverflow.com/questions/78126365/unable-to-pg-dump-from-supabase-project) useful as it seems you need to be using Postgres 16 locally. Was able to fix with `brew install postgresql@16` and `brew link --overwrite postgresql@16`

**Comment by acylicsalmon** [comment]
connecting via R, I found the answer using the following:

```
con <- dbConnect(RPostgres::Postgres(),
                 dbname = "postgres",
                 gssencmode = 'disable',
                 host = "aws-0-us-east-1.pooler.supabase.com",
                 port = 5432, ....
```

specifically the `gssencmode = 'disable',`

**Comment by div-dhingra** [comment]
> connecting via R, I found the answer using the following:
> 
> ```
> con <- dbConnect(RPostgres::Postgres(),
>                  dbname = "postgres",
>                  gssencmode = 'disable',
>                  host = "aws-0-us-east-1.pooler.supabase.com",
>                  port = 5432, ....
> ```
> 
> specifically the `gssencmode = 'disable',`

Can 2nd that adding 'gssencmode = 'disable' to the connection parameters/string fixes the issue in Flask (Python). Not sure if it would fall under a security-vulnerability, however.

**Comment by TheBuilderJR** [comment]
+1

**Comment by bdogrep** [comment]
+1

**Comment by jiahuijiang** [comment]
I'm seeing this with Supabase CLI even after upgrading my local postgresql to 16.

**Comment by L422Y** [comment]
> I'm seeing this with Supabase CLI even after upgrading my local postgresql to 16.

Not sure if this will help - but I had to rebuild some things and make sure they were using the right libraries, instead of 14... maybe we need to reinstall the CLI? idk just thought I'd provide this:

```sh
export PKG_CONFIG_PATH=/opt/homebrew/Cellar/postgresql@16/16.6/lib/pkgconfig
make install
```

**Comment by erpview** [comment]
well still got the same issue as above notihing works

**Comment by erpview** [comment]
has anyone an idea to fix it o make full dump of the project?

**Comment by bdogrep** [comment]
Even after upgrading PostgreSQL to version 16, your system might still be using older versions internally.

One reliable solution is to remove all PostgreSQL versions except version 16 - this approach resolved the issue in my environment.

> **Important**: If you have existing databases, make sure to backup your data before proceeding.

**Comment by ps6067966** [comment]
Facing same issue

**Comment by LemuelKL** [comment]
I fixed this issue on my system by upgrading postgreSQL 14.5 to 17.2. I'm on MacOS 15.2 (24C101) and using homebrew.

```bash
> brew install postgresql@17
> psql -V
psql (PostgreSQL) 17.2 (Homebrew)
```

**Comment by NestorXX-XX** [comment]
# GSSAPI Negotiation Error in Supabase PostgreSQL Connections (SOLUTION)

## Problem
You may encounter:
```
connection to server at "<host>" (<IP>), port <port> failed: received invalid response to GSSAPI negotiation
```
This occurs due to GSSAPI encryption incompatibility with Supabase's pooler.

---

## Solution
Add `gssencmode=disable` to your connection settings.

### Connection Examples

**Django**
```python
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql_psycopg2',
        'NAME': 'postgres',
        'USER': '<user>',
        'PASSWORD': '<password>',
        'HOST': 'aws-<region>.pooler.supabase.com',
        'PORT': '6543',
        'OPTIONS': {'gssencmode': 'disable'},
        'CONN_MAX_AGE': 500,
    }
}
```

**SQLAlchemy**
```python
engine = create_engine("postgresql+psycopg2://<user>:<password>@aws-<region>.pooler.supabase.com:6543/<dbname>?gssencmode=disable")
```

**psql Command**
```bash
psql "host=aws-<region>.pooler.supabase.com port=6543 dbname=postgres user=<user> password=<password> gssencmode=disable"
```

---

## Upgrade PostgreSQL Client
Update PostgreSQL client to version 16+:
```bash
brew install postgresql@16
brew link --overwrite postgresql@16
psql -V
```

---

## References
- [PostgreSQL Documentation: GSSAPI](https://www.postgresql.org/docs/current/gssapi.html)
- [Supabase Documentation](https://supabase.com/docs)

**Comment by andrebattagello** [comment]
followed @NestorXX-XX suggestion (using `psql [host=...]` and it worked for me

**Comment by JosephNK** [comment]
same here

**Comment by tessaneal** [comment]
> 'gssencmode': 'disable'

This worked for me!

**Comment by henrywithu** [comment]
> specifically the `gssencmode = 'disable',`

tyvm my man

**Comment by pranavmappoli** [comment]
Now its 17

```
brew install postgresql@17
brew link --overwrite postgresql@17
```

and use `gssencmode=disable`

**Comment by Hallidayo** [comment]
Hi all, I'm going to move this over as a discussion and thank you all for your help on this one!

**Comment by rxMeet-2960** [comment]
```python
conn = psycopg2.connect(
      host=host,
      port=port,
      database=database,
      user=user,
      password=password,
      gssencmode='disable'
 )
 ``` 
 Worked in python however
I cannot connect it with DBT or Soda despite passing gssencode: disable in yaml files. If anyone can lend me some help
